tensorflow/compiler/mlir/tensorflow/transforms/decompose_resource_ops.td

```diff
@@ -348,7 +351,8 @@ def DecomposeResourceScatterAdd : Pat<
     $resource,
     (TF_TensorScatterAddOp
       (CreateTFReadVariableOp $src_op, $updates, $resource),
-      $indices,
+      (TF_ExpandDimsOp $indices,
```

Comment:
We will need to expand dims for all resource scatter ops. Could you do that?

Reply:
on it! :)

Reply:
Currently there only exist ResourceScatterAddOp and ResourceScatterUpdateOp. Should we just do these two, or should we add other decompositions for resourcescatterops here as well:
https://github.com/tensorflow/tensorflow/blob/270104f2f6e0136531ceda4922c95eb501f70d7c/tensorflow/compiler/tf2xla/kernels/variable_ops.cc#L198 
L198 to L302. And if we need to do it for ResourceScatterNdOps as well, is it the same pattern? 🤔

Reply:
Doing for just these two should do.

Reply:
> > Please update PR title, add issue description along with a rationale of the fix.
> 
> I still see "Handle indices of rank-1 in Scatter Ops" here.

haha sorry yeah, I was waiting for this answer before I upload again S.T we wont trigger the builder unnecessarily :)

Reply:
Okay, now it's done! :)